src/impurity_init.ipp

```diff
@@ -362,7 +362,7 @@ void HybridizationSimulation<IMP_MODEL>::read_eq_time_two_particle_greens_meas()
     return;
   }
 
-  std::ifstream infile_f(boost::lexical_cast<std::string>(par[fname_key]).c_str());
+  std::ifstream infile_f(boost::lexical_cast<std::string>(par[fname_key].template as<std::string>()));
```

Comment:
`lexical_cast` is not needed here (and actually was erroneous, to begin with!). `boost::lexical_cast<string>` works by outputting its argument into a stream; however, there is no guarantee that the resulting string is a value of the argument in any sense: it's just its string representation. (That is, conversion from string representation back to the value may not be correct or at all possible; in this case, a parameter is represented as a string containing, for debugging purposes, its name, value and type.  I am going to change this, see  issue https://github.com/ALPSCore/ALPSCore/issues/490). This code was working with a previous version of parameters accidentally. What is needed here not a lexical cast, but a _static cast_: 

Correct code could be either 
```c++
std::ifstream infile_f(par[fname_key].as<std::string>());
```
or 
```c++
std::ifstream infile_f(static_cast<std::string>(par[fname_key]));
```
or simply
```c++
std::ifstream infile_f(std::string(par[fname_key]));
```

(i do not think that `template` is needed here).